src/node_http2.cc

```diff
@@ -763,6 +763,14 @@ void Http2Session::Close(uint32_t code, bool socket_closed) {
 
   flags_ |= SESSION_STATE_CLOSED;
 
+  // If we are writing we will get to make the callback in OnStreamAfterWrite.
+  if ((flags_ & SESSION_STATE_WRITE_IN_PROGRESS) == 0) {
+    Debug(this, "make done session callback");
+    HandleScope scope(env()->isolate());
+    MakeCallback(env()->ondone_string(), 0, nullptr);
+    if (stream_ != nullptr) stream_->ReadStart();
```

Comment:
Also, this doesn't unset `SESSION_STATE_READING_STOPPED` from `flags_` though I'm not sure if we should do it as we are basically done with the session. Perhaps we should at least for correctness?

Reply:
Hm, why do we start reading again here? But yes, I think this should unset that flag as we do use that to keep track of whether we’re reading or not

Reply:
That is in order to actually receive the FIN packet from the other party after `.end()`.
> Now after the session is closed we call ReadStart() on the underlying
stream to allow socket to receive the remaining data and FIN packet.
Previously only ReadStop() was used therefore blocking the receival of
FIN by the socket and 'end' event after .end() call.

The handle will be stopped by the `.destroy()` in the `net`'s `_final` after we receive the FIN packet.